### ISSUE TITLE
Generate .appsody-config.yaml when creating the project

### ISSUE THREAD

**Issue by kewegner** [appsody/appsody#35]
Per @ebullient 

.appsody-config.yaml is currently checked into the starters inside the stack. This creates technical debt for the stack, as the .appsody-config.yaml in each starter in the stack has to also be updated when the version changes.

The .appsody-config.yaml file should be generated at project create time to use the version of the stack that created it.

**Comment by tomleah** [comment]
Generating a `.appsody-config.yaml` can be easily achieved. We should do this with the CLI when using `appsody init`. 

There was a discussion on doing this in the CI process when building the `.tgz.gz` for the template but we decided this is not the way to go as it prevents stack creators.

**Comment by ebullient** [comment]
prevents stack creators from doing what.. from determining what to put in that file for versioning?

**Comment by tomleah** [comment]
As `.appsody-config.yaml` is required by Appsody it prevents stack creators from being able to test/develop the stack, as the `.appsody-config.yaml` would only exist after the stack has gone through CI. 

They could run the CI script locally to generate the file but I think it would be easier to be built into the CLI itself.

**Comment by neeraj-laad** [comment]
We are documenting that stack providers run our build scripts to get updated docker images, template archives and a local index so that they can test the full appsody capabilities including `init`.

If they are going this to test locally then adding this to the CI makes it a much simpler solution.

**Comment by tomleah** [comment]
Of course, I didn't consider that stack providers would already be running the CI scripts so this makes more sense 🙂
